pkg/controllermanager/controller/shoot/maintenance/reconciler.go

```diff
@@ -150,12 +162,58 @@ func (r *Reconciler) reconcile(ctx context.Context, log logr.Logger, shoot *gard
 		reasonsForWorkerPoolKubernetesUpdate[w.Name] = reasonForWorkerPoolKubernetesUpdate
 	}
 
-	maintainOperation(shoot)
+	for workerPool, reason := range reasonsForWorkerPoolKubernetesUpdate {
+		if reason != "" {
+			operation := fmt.Sprintf("For worker pool %s: %s", workerPool, reason)
+			operations = append(operations, operation)
+		}
+	}
+
+	operation := maintainOperation(maintainedShoot)
+	if operation != "" {
+		operations = append(operations, "Other task started during maintenance time - "+operation)
+	}
+
+	if operations != nil {
+		patch := client.MergeFrom(shoot.DeepCopy())
+		shoot.Status.LastMaintenance = &gardencorev1beta1.LastMaintenance{
+			Description:   strings.Join(operations, ", "),
+			TriggeredTime: metav1.Time{Time: r.Clock.Now()},
+		}
+
+		// First dry run the update call to check if it can be executed successfully.
+		// If not shoot maintenance is marked as failed and is retried only in
+		// next maintenance window.
+		if err := r.Client.Update(ctx, maintainedShoot, &client.UpdateOptions{
+			DryRun: []string{metav1.DryRunAll},
+		}); err != nil {
+			if hasMaintainNowAnnotation(shoot) {
+				delete(shoot.Annotations, v1beta1constants.GardenerOperation)
+			}
+			shoot.Status.LastMaintenance.Status = pointer.String(fmt.Sprintf("Maintenance failed: %s", err.Error()))
+			if err := r.Client.Status().Patch(ctx, shoot, patch); err != nil {
+				log.Error(err, "Failed to patch shoot status")
+				return err
```

Comment:
```suggestion
				return err
```

From https://github.com/gardener/gardener/blob/master/docs/development/logging.md#reconciler-loggers:

> Hence, don't log an error that you are returning.